mm2src/common/custom_futures/repeatable.rs

```diff
@@ -236,6 +236,17 @@ where
         self.repeat_every(Duration::from_secs_f64(repeat_every))
     }
 
+    /// Repeat the future until it's ready.
+    ///
+    /// # Warning
+    ///
+    /// This may lead to an endless loop if the future is never ready.
```

Comment:
This seems pretty dangerous tbh. In this case we should have some default timeout functionality to overcome this risk.

Reply:
I agree with you. I moved and refactor the following implementation to a separate method
https://github.com/KomodoPlatform/atomicDEX-API/blob/9350a3ef852a896685243be53f21d6c8cd4769b2/mm2src/coins/eth.rs#L1672-L1692

I'm really not sure if we can change this logic, @artemii235 what do you think if we add a timeout?